test/NuGet.Core.Tests/NuGet.Protocol.Tests/HttpHandlerResourceV3ProviderTests.cs

```diff
@@ -124,5 +125,45 @@ static IEnumerable<DelegatingHandler> GetDelegatingHandlers(HttpMessageHandler h
                 }
             }
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task TryCreate_WhenCertificateValidationIsConfigured_HandlerShouldReflectConfiguration(bool disableCertificateValidation)
+        {
+            // Arrange
+            Mock<IProxyCache> proxyCache = new();
+            proxyCache.Setup(pc => pc.GetProxy(It.IsAny<Uri>())).Returns((IWebProxy)null);
+
+            PackageSource packageSource = new(_testPackageSourceURL, "source")
+            {
+                DisableTLSCertificateValidation = disableCertificateValidation
+            };
+
+            SourceRepository sourceRepository = new(packageSource, Array.Empty<INuGetResourceProvider>());
+
+            HttpHandlerResourceV3Provider target = new(proxyCache.Object);
+
+            // Act
+            var result = await target.TryCreate(sourceRepository, CancellationToken.None);
+
+            // Assert
+            result.Item1.Should().BeTrue();
+            HttpHandlerResourceV3 resource = (HttpHandlerResourceV3)result.Item2;
+            resource.Should().NotBeNull();
+
+            HttpClientHandler clientHandler = resource.ClientHandler;
+
+            if (disableCertificateValidation)
+            {
+                clientHandler.ServerCertificateCustomValidationCallback.Should().NotBeNull();
+                var callbackResult = clientHandler.ServerCertificateCustomValidationCallback.Invoke(null, null, null, SslPolicyErrors.RemoteCertificateChainErrors);
```

Comment:
This test is specifying a `SslPolicyErrors`, which makes me wonder if the implementation should check for that?

Reply:
I was testing the client handler to ignore the errors. Since for the package source we have set `DisableTLSCertificateValidation` to `true`, it should be ignored and the test checks for that.

Reply:
Could pass in all the values instead of just one:

```suggestion
                var callbackResult = clientHandler.ServerCertificateCustomValidationCallback.Invoke(null, null, null, 
                    SslPolicyErrors.RemoteCertificateNotAvailable
                        & SslPolicyErrors.RemoteCertificateNameMismatch
                        & SslPolicyErrors.RemoteCertificateChainErrors);
```

Reply:
Nice one. Fixed